rules/react.js

```diff
@@ -104,5 +104,195 @@ module.exports = {
+        reservedFirst: true,
+      },
+    ],
+
+    // ==========================
+    // ! IMPORTANT: CAN'T use --fix *
+    // ! Enable with precaution
+    // ==========================
+    // https://github.com/yannickcr/eslint-plugin-react/blob/843d71a432baf0f01f598d7cf1eea75ad6896e4b/docs/rules/jsx-sort-default-props.md
+    // AirBnB
+    // 'react/jsx-sort-default-props': ['off', {
+    //   ignoreCase: true,
+    // }],
+    // ============
+    // Proposed:
+    // In style with sorting component's props
+    // 'react/jsx-sort-default-props': [
+    //   'error',
+    //   {
+    //     ignoreCase: true,
+    //   },
+    // ],
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-direct-mutation-state.md
+    // AirBnB
+    // 'react/no-direct-mutation-state': 'off',
+    // ============
+    // Proposed:
+    // Enforce immutability
+    'react/no-direct-mutation-state': 'error',
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md
+    // AirBnB
+    // 'react/forbid-component-props': ['off', { forbid: [] }],
+    // ============
+    // Default value: className & style
+    // Proposed:
+    // Enforce styling inside styled-component rather than class
+    // We can pass custom props to styled-component and assign logic there
+    // Span allowed for our icon set like linear icons but we can also specify the required icon
+    // inside styled-component attrs but that means we need to create a lot of icon component.
+    // If no additional styling required, using className for span icon probably good enough.
+    // Also forbid using "style" as component's props.
+    'react/forbid-component-props': [
+      'error',
+      {
+        forbid: [{ propName: 'className', allowedFor: ['span'] }, { propName: 'style' }],
+      },
+    ],
+
+    // https://eslint.org/docs/rules/max-statements-per-line
+    // AirBnB
+    // 'max-statements-per-line': ['off', { max: 1 }],
+    // ============
+    // Proposed:
+    // max only 1 statement per line
+    // GOOD:
+    // if (true) { doThis(); }
+    // ERROR:
+    // if (true) { doThis(); doThat(); }
+    'max-statements-per-line': ['error', { max: 1 }],
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md
+    // AirBnB
+    // 'react/jsx-no-literals': ['off', { noStrings: true }],
+    // ============
+    // Proposed:
+    // Always wrap sting inside literal
+    // GOOD:
+    // <span>{'Hello'}</span>
+    // BAD:
+    // <span>Hello</span>
+    // WHY: consistency and make it more specific to JSX's literal rather than string literal
+    'react/jsx-no-literals': ['error', { noStrings: true }],
+
+    // ==============================================================================
+    // ==============================================================================
+    // TODO: Enable when moved to prettier-eslint package instead of current config
```

Comment:
Can we move these comments out and into another PR.
We should not commit commented code

Reply:
Actually I think we should make a separate PR. Branches are for code, not comments